### PR TITLE
Added template overrides

### LINKED ISSUE #31021: Styling of recovery codes seems wrong
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

login/ui

### Describe the bug

The style of the recovery codes download 

### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

look good

### Actual behavior

doesn't

### How to Reproduce?

enable recovery codes feature and download the codes from the account console

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by edewit**
fixes: #31021

### CODE REVIEW

**Comment by edewit** [comment]
image
image

**Comment by edewit** [comment]
@xianli123 right, made "cancel button" a link

**Comment by thomasdarimont** [comment]
@edewit I think you could also fix the invalid css selector in the javascript snippet, 
`.kc-recovery-codes-list li ` should be `#kc-recovery-codes-list li `.

See: https://github.com/keycloak/keycloak/issues/31260

File: themes/src/main/resources/theme/base/login/login-recovery-authn-code-config.ftl [hunk 1]
```diff
@@ -5,182 +5,188 @@
...
+        hour: 'numeric',
+        minute: 'numeric',
+        timeZoneName: 'short'
+    };
+
+    return dt.toLocaleString('en-US', options);
+}
+
+function parseRecoveryCodeList() {
+    var recoveryCodes = document.querySelectorAll(".kc-recovery-codes-list li");
```
Comment (thomasdarimont) [inline themes/src/main/resources/theme/base/login/login-recovery-authn-code-config.ftl:115]:
```suggestion
    var recoveryCodes = document.querySelectorAll("#kc-recovery-codes-list li");
```

File: themes/src/main/resources/theme/keycloak.v2/login/login-recovery-authn-code-config.ftl [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        hour: 'numeric',
+        minute: 'numeric',
+        timeZoneName: 'short'
+    };
+
+    return dt.toLocaleString('en-US', options);
+}
+
+function parseRecoveryCodeList() {
+    var recoveryCodes = document.querySelectorAll(".kc-recovery-codes-list li");
```
Comment (thomasdarimont) [inline themes/src/main/resources/theme/keycloak.v2/login/login-recovery-authn-code-config.ftl:115]:
```suggestion
    var recoveryCodes = document.querySelectorAll("#kc-recovery-codes-list li");
```

File: themes/src/main/resources/theme/keycloak.v2/login/resources/css/styles.css [hunk 3]
```diff
@@ -69,3 +69,10 @@ div.kc-logo-text span {
     border-color: transparent black transparent transparent;
 }
 
+.kc-recovery-codes-list {
```
Comment (thomasdarimont) [inline themes/src/main/resources/theme/keycloak.v2/login/resources/css/styles.css:72]:
```suggestion
#kc-recovery-codes-list {
```

**Review by thomasdarimont — CHANGES_REQUESTED** [review]
We should use the proper css selector for the recovery-codes-list.

File: themes/src/main/resources/theme/base/login/login-recovery-authn-code-config.ftl
Comment (jonkoops) [inline themes/src/main/resources/theme/base/login/login-recovery-authn-code-config.ftl:1]: I see a bunch of PatternFly 5 classes here. The base theme is used by login v1 as well, so would this not break that version of the login theme as it doesn't include PatternFly 5?

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: themes/src/main/resources/theme/base/login/login-recovery-authn-code-config.ftl
Comment (edewit) [reply themes/src/main/resources/theme/base/login/login-recovery-authn-code-config.ftl:1]: v1 has it's own version of this

Comment (jonkoops) [reply themes/src/main/resources/theme/base/login/login-recovery-authn-code-config.ftl:1]: Still, users could be extending the `base` theme, that happens quite frequently. If the V2 one has it's own template this one does not need to be updated.

**Review by jonkoops — APPROVED** [review]

**Review by thomasdarimont — APPROVED** [review]

**Pull request merged**